src/shared/creds-util.c

```diff
@@ -83,6 +83,40 @@ int read_credential(const char *name, void **ret, size_t *ret_size) {
                         (char**) ret, ret_size);
 }
 
+int get_credential_user_password(const char *username, char **password, bool *is_hashed)
+{
```

Comment:
please follow coding style, i.e. put { on the line as function name, i.e. remove one newline here